odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+        """The source position function.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (`ndim`,)
+
+            The source position at ``z`` on the circle with radius ``r`` at
+            a given rotation angle :math:`\\phi`, defined as :math:`(r
+            \\sin\\phi, -r\\cos\\phi, 0)`
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        angle += self.angle_offset
+
+        # ASTRA cone_vec convention
+        return self.src_radius * np.array([sin(angle), -cos(angle), 0])
+
+        # TODO: backprojection weighting function?
+
+
+class HelicalConeFlatGeometry(ConeFlatGeometry):
+    """Helical cone beam geometry in 3d with flat detector.
+
+    The source moves on a circle with radius ``r``, and the detector
+    reference point is opposite to the source on a circle with radius ``R``
+    and aligned tangential to the circle.
+
+    The motion parameter is the (1d) rotation angle parametrizing
+    source and detector positions.
+    """
+
+    def __init__(self, angle_intvl, dparams, src_rad, det_rad,
+                 spiral_pitch_factor, agrid=None, dgrid=None,
+                 angle_offset=0, axis=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        angle_intvl : `Interval` or 1-dim. `IntervalProd`
+            The motion parameters
+        dparams : `Rectangle` or 2-dim. `IntervalProd`
+            The detector parameters
+        src_rad : `float`
+            Radius of the source circle, must be positive
+        det_rad : `float`
+            Radius of the detector circle, must be positive
+        spiral_pitch_factor : `float`
+            Dimensionless factor given by the table feed per rotation
+            divided by the total collimation width. The total collimation
+            width is considered at isocenter and  given by the slice thickness
+            for a single-slice spiral acquisition and by the number of
+            detector rows times the slice thickness for multiple slice
+            spiral acquisition.
+        agrid : 1-dim. `TensorGrid`, optional
+            A sampling grid for `angle_intvl`. Default: `None`
+        dgrid : 2-dim. `TensorGrid`, optional
+            A sampling grid for `dparams`. Default: `None`
+        angle_offset : `float`, optional
+            Offset to the rotation angle in the azimuthal plane. Does not
+            affect the offset in the longitudinal direction. Default: 0
+        axis : `int` or 3-element array, optional
+            Defines the rotation axis via a 3-element vector or a single
+            integer referring to a standard axis. Default: `None`
```

Comment:
I actually have to agree even though I'm guilty of this option. The potential problem is the "hard wired" connection between data axes 0, 1, 2 referring to first, second and third indices of an array, and spatial axes usually called x, y, z. In principle there is no connection, nobody keeps (or should keep) people from letting the spatial z axis corresponding to the first data axis. By specifying `axis` as a spatial vector, it is clear what is meant, while an index could be both. And yes, `[0, 0, 1]` is short enough.